### DISCUSSION TITLE
iOS purchaseUpdatedListener called multiple times

### DISCUSSION THREAD

**Discussion by Saikedo** [hyochan/react-native-iap#2707] [General]
Hi, can someone please explain how to deal with subscriptions on iOS where purchaseUpdatedListener gets called multiple times? When I allow some time to pass and then call requestSubscription,  I end up getting bunch of old renewal purchases arriving through purchaseUpdatedListener. This makes it almost impossible for me to rely on the incoming updates to update my UX for example because I might end up receiving like 5 back to back updates.

I found something called clearTransactionIOS but I was not sure how that can be relevant here especially since the example says to not use it in production code. 

Can anyone please advice on how this should be handled ? Thank you very much.

-----------------

It actually seems like getSubscriptions is the one that triggers all the old subscriptions to arrive to purchaseUpdateListerner. Is that expected ?

**Comment by hyochan** [comment]
Thank you for using react-native-iap!

**Versions below v14 are no longer supported.** We cannot provide support or bug fixes for v12/v13 and earlier versions.

## Why Upgrade?

v14+ brings significant improvements:

- **Nitro Modules** - JSI-based native bridge for better performance
- **StoreKit 2** - Modern iOS in-app purchase API (iOS 15+)
- **Google Play Billing 7.0+** - Latest Android billing features
- **Unified API** - Cleaner, more consistent interface
- **Better error handling** - Unified `ErrorCode` enum across platforms
- **Active maintenance** - Bug fixes and new features

## Migration Guide

Please follow the migration guide to upgrade:

👉 **[Migration Guide (v13 to v14)](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14)**

## Requirements

- **iOS**: 15.0+
- **Xcode**: 16.4+
- **React Native**: 0.71+
- **Android**: Google Play Billing 7.0+

## Need Help?

If you encounter issues during migration, please:

1. Read the [Migration Guide](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14) thoroughly
2. Check the [Troubleshooting Guide](https://hyochan.github.io/react-native-iap/docs/guides/troubleshooting)
3. Open a new issue with v14+ if problems persist
